components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksUseCases.kt

```diff
@@ -190,14 +191,21 @@ class AppLinksUseCases(
     inner class OpenAppLinkRedirect internal constructor(
         private val context: Context
     ) {
-        operator fun invoke(appIntent: Intent?, failedToLaunchAction: () -> Unit = {}) {
+        operator fun invoke(
+            appIntent: Intent?,
+            launchInNewTask: Boolean = true,
```

Comment:
Do we want the default `launchInNewTask` for redirect to be true here? 🤔

Reply:
Yes, in most cases we should be launching intents in new tasks.  Since to launch another app without the `FLAG_ACTIVITY_NEW_TASK` flag is usually not what we want.

I added the parameter here so if the caller knows a very specific case then it can set it to false.  But the default behaviour should be launching intents in new task.

Reply:
Okay, that makes sense.